examples/hello/hello.c

```diff
@@ -6,22 +6,18 @@
 static int begin_request_handler(struct mg_connection *conn)
 {
     const struct mg_request_info *request_info = mg_get_request_info(conn);
-    char content[100];
 
     // Prepare the message we're going to send
-    int content_length = snprintf(content, sizeof(content),
-                                  "Hello from civetweb! Remote port: %d",
-                                  request_info->remote_port);
 
     // Send HTTP reply to the client
     mg_printf(conn,
               "HTTP/1.1 200 OK\r\n"
               "Content-Type: text/plain\r\n"
-              "Content-Length: %d\r\n"        // Always set Content-Length
-              "\r\n"
-              "%s",
-              content_length, content);
+              "transfer-encoding: chunked\r\n"
+              "\r\n");
 
+     mg_printf(conn, "Hello from civetweb! Remote port: %d", request_info->remote_port);
```

Comment:
For chunked tranfer, you must not just send the data, but first you need to send the following chunk size (in hex).
e.g.

    int content_length = snprintf(content, sizeof(content),
                                  "Hello from civetweb! Remote port: %d",
                                  request_info->remote_port);
    mg_printf(conn, "%x\r\n", content_length);
    mg_printf(conn, "%s\r\n", content);  /* or better mg_write, especially for binary data */

then you must send an empty chunk to let the client know it's complete

    mg_printf(conn, "0\r\n\r\n");

By itself, mg_printf does not know if you are sending a header or a part of the body, and if this body needs chunked encoding or not.